gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,260 @@
+) -> hl.MatrixTable:
+    """
+    Generate the QC MT per specified population and generate PCA info.
+
+    :param mt: The QC MT output by the 'compute_subpop_qc_mt' function
+    :param pop: Population to which the Matrix Table should be filtered
+    :param min_af: Minimum population variant allele frequency to retain variant in QC MT
+    :param min_inbreeding_coeff_threshold: Minimum site inbreeding coefficient to retain variant in QC MT
+    :param release: Whether or not to filter to only release samples
+    :param high_quality: Whether or not to filter to only high quality samples
+    :param outliers: Table keyed by column containing outliers to remove
+    :param ht_read_if_exists: Whether or not to read an existing Table of PCA data if it exists
+    :param outlier_string: String used to describe which outliers(if any) were removed
+    :return: Table with sample metadata and PCA scores for the specified population to use for subpop analysis
+    """
+    meta_ht = meta.ht()
+    relateds = pca_related_samples_to_drop.ht()
+    hard_filtered_ht = hard_filtered_samples.ht()
+
+    # Add info to the MT
+    info_ht = get_info(split=False).ht()
+    info_ht = info_ht.annotate(
+        info=info_ht.info.select(
+            # No need for AS_annotations since it's bi-allelic sites only
+            **{x: info_ht.info[x] for x in info_ht.info if not x.startswith("AS_")}
+        )
+    )
+    mt = mt.annotate_rows(info=info_ht[mt.row_key].info)
+
+    # Add sample metadata to the QC MT
+    mt = mt.annotate_cols(**meta_ht[mt.col_key])
+
+    # Filter the  QC MT to the specified pop
+    pop_mt = mt.filter_cols(mt.pop == pop)
+
+    # Apply specified filters
+    if remove_hard_filtered_samples:
+        pop_mt = pop_mt.filter_cols(hl.is_missing(hard_filtered_ht[pop_mt.col_key]))
+    if release:
+        pop_mt = pop_mt.filter_cols(pop_mt.release)
+    if high_quality:
+        pop_mt = pop_mt.filter_cols(pop_mt.high_quality)
+    if outliers is not None:
+        pop_mt = pop_mt.filter_cols(hl.is_missing(outliers[pop_mt.col_key]))
+
+    pop_mt = pop_mt.filter_rows(hl.agg.any(pop_mt.GT.is_non_ref()))
+
+    # Generate a QC MT for the given pop
+    pop_qc_mt = get_qc_mt(
+        pop_mt,
+        min_af=min_af,
+        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+        min_hardy_weinberg_threshold=None,
+        ld_r2=None,
+        filter_lcr=False,
+        filter_decoy=False,
+        filter_segdup=False,
+    )
+
+    # Generate PCA data
+    release_string = "_release" if release else ""
+    high_quality_string = "_high_quality" if high_quality else ""
+    ht_path = (
+        get_sample_qc_root()
+        + f"/subpop_analysis/{pop}/{pop}_scores{release_string}{high_quality_string}{outlier_string}.ht"
+    )
+
+    if not (ht_read_if_exists and file_exists(ht_path)):
+        pop_pca_evals, pop_pca_scores, pop_pca_loadings = run_pca_with_relateds(
+            pop_qc_mt, relateds
+        )
```

Comment:
I just got thinking that we probably want to save all three of these that are returned. I wonder if this should be split out of the function. The first gets the pop specific QC MT and returns it. Then this section can just go into main, and write PCA info that is returned, along with this HT (or don't even write this HT out, just have a function that annotates the scores HT). Thoughts?